pkg/controllers/management/clusterdeploy/clusterdeploy.go

```diff
@@ -101,11 +113,17 @@ func (cd *clusterDeploy) doSync(cluster *v3.Cluster) error {
 	if err != nil {
 		return err
 	}
+
 	err = cd.deployAgent(cluster)
 	if err != nil {
 		return err
 	}
 
+	err = cd.validateAgent(cluster)
```

Comment:
Could this be called before deployagent then the correct one would be deployed in this run? I don't follow returning an error when it really isn't an error. If someone sets the override the controller should respect that and not throw an error in the logs.